Gordon360/Services/NewsService.cs

```diff
@@ -153,9 +173,156 @@ public StudentNewsViewModel EditPosting(int newsID, StudentNews newData)
+                string base64Data = System.Convert.ToBase64String(imageBytes);
+                return base64Data;
+            }
+        }
+
+        /// <summary>
+        /// Iterates through a collection of news items that have just been queried from
+        /// the database, replacing the Image entry of each (which contains the path to an image)
+        /// with the image itself. This is done because when the frontend makes requests for news items,
+        /// it needs the image data and not the path, but we store the path in the DB for efficiency and
+        /// storage reasons.
+        /// </summary>
+        /// <param name="items">The collection of student news entries</param>
+        /// <returns>The collection of entries with base64 data in their image variable</returns>
+        private IEnumerable<StudentNewsViewModel> InsertImageData(IEnumerable<StudentNewsViewModel> items)
+        {
+            foreach (StudentNewsViewModel item in items)
+            {
+                string imagePath = item.Image;
+
+                if (imagePath != null)
+                {
+                    item.Image = GetBase64ImageDataFromPath(imagePath);
+                }   
+            }
+            return items;
+        }
+
+        /// <summary>
+        /// Uploads a news image
+        /// Can be used to add an image to a new posting or to replace an image
+        /// for an existing posting.
+        /// </summary>
+        /// <param name="imageData">The base64 image data to be stored</param>
+        /// <param name="snid">The SNID of the news item to which the image belongs</param>
+        private void UploadImage(string imageData, int snid)
+        {
+            if (imageData == null) { return; }
+
+            var newsPost = Get(snid);
+
+            if (!System.IO.Directory.Exists(FolderPath))
+            {
+                System.IO.Directory.CreateDirectory(FolderPath);
+            }
+
+            string fileName = snid + ".jpg";
+            string imagePath = FolderPath + fileName;
+
+            byte[] imageDataArray = System.Convert.FromBase64String(imageData);
+
+            try
+            {
+                //First, if the Image path is empty- a new post- and the imageData is empty,
+                //It means a news submission was made without a picture, so we won't uploade anything.
+                // We'll just return.
+                if (imageData == null && newsPost.Image == null)
+                {
+                    return;
+                }
+
+                System.Diagnostics.Debug.WriteLine(newsPost.Image + " is value ");
+                //If a new post, Image path will be null. If there is image data,
+                //we need to conitune, so first we add the path to the DB column. 
+                if (newsPost.Image == null && imageData != null)
+                {
+                    newsPost.Image = imagePath;
+                }
+
+                _unitOfWork.Save();
```

Comment:
It's a little bit odd to me that we're tightly coupling uploading the image to the image file server with saving the image path to the database. I think it would be much clearer if you had this method take the `imageData` and the `imagePath` and be only responsible for uploading the image to the server (as it's name suggests), and then you have a different method that is only responsible for saving the image path to the database. Then, the calling code would look like:
```
var imagePath = FolderPath + snid + ".jpg";
UploadImage(imageData, imagePath);
SaveImagePath(imagePath);
```

And of course, if `UploadImage` encounters a problem, it would throw an exception, which would prevent saving an invalid path to the database.